### PR TITLE
Ensure the something incorrect disclosure is shown

### PR DESCRIPTION
**Pull request by Badlapje**
Prior to this change, the id for the input element toggling the tooltip was generated incorrectly.  This made all labels of different organisations toggle the same disclosure modal.

This change uses the organizationId as an id for the input.  Thus ensuring that the modal is toggled correctly.

Pivotal ticket: https://www.pivotaltracker.com/story/show/180095824

### CODE REVIEW

File: src/OpenConext/ProfileBundle/Resources/views/MyServices/AttributeList/idp.html.twig [hunk 1]
```diff
@@ -73,16 +73,13 @@
             <a class="idpRow__supportUrl" href="{{ 'profile.my_services.idpRow.support_url'|trans }}" target="_blank" rel="noreferrer noopener">{{ 'profile.my_services.idpRow.support_link'|trans }}</a>
         </div>
     {% else %}
-        {% set id %}
-            incorrect_{{ sourceName }}{{ loop.index }}
-        {% endset %}
-        <input type="checkbox" tabindex="-1" aria-hidden="true" class="modal visually-hidden" id="{{ id|trim }}" name="{{ id|trim }}" />
+        <input type="checkbox" tabindex="-1" aria-hidden="true" class="modal visually-hidden" id="{{ orgId|trim }}" name="{{ orgId|trim }}" />
```
[the remark this answers is not in the document]
Comment (Badlapje) [reply src/OpenConext/ProfileBundle/Resources/views/MyServices/AttributeList/idp.html.twig:76]: I don't think this is needed.  I could use the filter anyway from a "if it doesn't hurt" perspective.  But i cannot find any docs on the HTML Attribute context which is mentioned in the above escape doc page to ensure it doesn't do damage.  

Looking at https://html.spec.whatwg.org/multipage/syntax.html#attributes-2, it seems clear that for attribute names escaping is needed (i assume that is what the escape docs mean with the html attribute context).  But for values everything goes, with the exception of an ambiguous ampersand (so in other words, don't use the html escape filter and it's ok).

The orgId contains either an url or a name but without the spaces ,which means effectively no chance for an ambiguous ampersand.

`{% set orgId = sp.entity.entityId.entityId|trim|split(' ')|join %}`

**Comment by Badlapje** [comment]
> 
> 
> In addition to the mention above, I also get this JS TypeError in my console:
> 
> ```
> Uncaught TypeError: r is null
>     781 https://profile.vm.openconext.org/build/app.ca4cc31e.js:1
>     781 https://profile.vm.openconext.org/build/app.ca4cc31e.js:1
> ```
> 
> Can u fix?

fixed 👍

File: src/OpenConext/ProfileBundle/Resources/views/MyServices/AttributeList/idp.html.twig (hunk 1 above)
[the remark this answers is not in the document]
Comment (MKodde) [reply src/OpenConext/ProfileBundle/Resources/views/MyServices/AttributeList/idp.html.twig:76]: Given the content is used in the `id` attribute, we do not run any potential security risks here. But from a html complience perspective, I feel an entity ID might contain unsafe characters to put into an attribute value. But maybe I'm exaggerating here. My concern is mainly that the ID could become useless when an exotic entity ID disrupts the attribute value.

When it comes to rules regarding escaping. I always like to read the OWASP cheat cheat: https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html#rule-2-attribute-encode-before-inserting-untrusted-data-into-html-common-attributes

An enitty Id should follow the guidelines described in [RFC2986](https://datatracker.ietf.org/doc/html/rfc3986#section-6) so putting it in an attribute value should pose little problems. But still, you never know what ends up there..

[the remark this answers is not in the document]
Comment (Badlapje) [reply src/OpenConext/ProfileBundle/Resources/views/MyServices/AttributeList/idp.html.twig:76]: Given that only ambiguous ampersands could cause problems, how's about we solve it this way:
change: 
`{% set orgId = sp.entity.entityId.entityId|trim|split(' ')|join %}` to: 
`{% set orgId = sp.entity.entityId.entityId|trim|split(' ')|join|split('&')|join %}`

[the remark this answers is not in the document]
Comment (Badlapje) [reply src/OpenConext/ProfileBundle/Resources/views/MyServices/AttributeList/idp.html.twig]: I added escaping using the escape filter, though I'm still deeply uneasy about adding a filter that is not documented to say which characters are escaped in which manner.  Seems to work locally, though given the uncertainty we cannot be sure it'll work with all values on production.

File: src/OpenConext/ProfileBundle/Resources/views/MyServices/AttributeList/idp.html.twig [hunk 2]
```diff
@@ -73,16 +73,14 @@
...
-        <input type="checkbox" tabindex="-1" aria-hidden="true" class="modal visually-hidden" id="{{ id|trim }}" name="{{ id|trim }}" />
+        {% set sanitizedId = orgId|trim|escape('html_attr') %}
+        <input type="checkbox" tabindex="-1" aria-hidden="true" class="modal visually-hidden" id="{{ sanitizedId }}" name="{{ sanitizedId }}" />
         {# note: the empty label is for display on mobile #}
         <p>{{ consentProvidedBy|raw }}<label tabindex="0" for="{{ id|trim }}" class="modal"></label>
         </p>
         {# note: the div is here only for IE11, as soon as IE11 support is scrapped, this can be removed along with the CSS/JS for it  #}
         <div class="ie11__label">
-            <label tabindex="0" for="{{ id|trim }}" class="modal">
+            <label tabindex="0" for="{{ orgId|trim }}" class="modal">
```
Comment (MKodde) [inline src/OpenConext/ProfileBundle/Resources/views/MyServices/AttributeList/idp.html.twig:83]: Mustn't this be `sanitizedId`. Forgive me for not diving into this myself.

**Review by MKodde — APPROVED** [review]
I think this'l do nicely. Save for one question that might be a false positive.

File: src/OpenConext/ProfileBundle/Resources/views/MyServices/AttributeList/idp.html.twig (hunk 2 above)
Comment (Badlapje) [reply src/OpenConext/ProfileBundle/Resources/views/MyServices/AttributeList/idp.html.twig]: quite right sir, eyes like a hawk!

fixed it 😄

File: src/OpenConext/ProfileBundle/Resources/views/MyServices/AttributeList/idp.html.twig [hunk 3]
```diff
@@ -73,13 +73,14 @@
...
-        <input type="checkbox" tabindex="-1" aria-hidden="true" class="modal visually-hidden" id="{{ orgId|trim }}" name="{{ orgId|trim }}" />
+        {% set sanitizedId = orgId|trim|escape('html_attr') %}
+        <input type="checkbox" tabindex="-1" aria-hidden="true" class="modal visually-hidden" id="{{ sanitizedId }}" name="{{ sanitizedId }}" />
         {# note: the empty label is for display on mobile #}
         <p>{{ consentProvidedBy|raw }}<label tabindex="0" for="{{ id|trim }}" class="modal"></label>
         </p>
         {# note: the div is here only for IE11, as soon as IE11 support is scrapped, this can be removed along with the CSS/JS for it  #}
         <div class="ie11__label">
-            <label tabindex="0" for="{{ orgId|trim }}" class="modal">
+            <label tabindex="0" for="{{ sanitizedId }}" class="modal">
```
Comment (MKodde) [inline src/OpenConext/ProfileBundle/Resources/views/MyServices/AttributeList/idp.html.twig:83]: Pico bello

**Review by MKodde — APPROVED** [review]

**Pull request merged**